src/main/java/loci/common/NIOFileHandle.java

```diff
@@ -105,6 +105,8 @@
   /** The original length of the file. */
   private Long defaultLength;
 
+  private long effectiveLength;
```

Comment:
Am I correct in assuming that the addition of a private field means this would break serialization when propagated up to the Bio-Formats stack

Reply:
Not necessarily - readers shouldn't be caching open file handles, so I wouldn't think instances of ```NIOFileHandle``` would be present in memo files.  For example:

https://github.com/openmicroscopy/bioformats/blob/develop/components/formats-api/src/loci/formats/FormatReader.java#L127

Probably merits a test before bumping the ```ome-common``` version number in Bio-Formats though, to be safe.